modules/lightning_features/lightning_api/modules/api_test/tests/src/Functional/EntityCRUTest.php

```diff
@@ -0,0 +1,178 @@
+  }
+
+  /**
+   * Tests create, read, and update of content and config entities via the
+   * API.
+   */
+  public function testEntities() {
+    $description = 'Created by Carl Linnaeus';
+    $data = [
+      'data' => [
+        'type' => 'taxonomy_vocabulary--taxonomy_vocabulary',
+        'id' => 'taxonomy_test_vocabulary',
+        'attributes' => [
+          'uuid' => 'taxonomy_test_vocabulary',
+          'name' => 'I\'m a vocab',
+          'vid' => 'im_a_vocab',
+          'description' => $description,
+          'status' => TRUE,
+        ]
+      ]
+    ];
+
+    // Create a taxonomy vocabulary (config entity).
+    $this->request('/jsonapi/taxonomy_vocabulary/taxonomy_vocabulary', 'post', $this->token, $data);
+
+    // Read the newly created vocabulary.
+    $response = $this->request('/jsonapi/taxonomy_vocabulary/taxonomy_vocabulary/taxonomy_test_vocabulary', 'get', $this->token);
+    $body = $this->decodeResponse($response);
+    $this->assertEquals($description, $body['data']['attributes']['description']);
+
+    $new_description = 'Refined by Johann Bartsch.';
+    $data = [
+      'data' => [
+        'id' => 'taxonomy_test_vocabulary',
+        'attributes' => [
+          'description' => $new_description,
+        ]
+      ]
+    ];
+
+    // Update the vocabulary.
+    $this->request('/jsonapi/taxonomy_vocabulary/taxonomy_vocabulary/taxonomy_test_vocabulary', 'patch', $this->token, $data);
+
+    // Read the updated vocabulary.
+    $response = $this->request('/jsonapi/taxonomy_vocabulary/taxonomy_vocabulary/taxonomy_test_vocabulary', 'get', $this->token);
+    $body = $this->decodeResponse($response);
+    $this->assertEquals('I\'m a vocab', $body['data']['attributes']['name']);
+    $this->assertEquals($new_description, $body['data']['attributes']['description']);
+
+    // Assert that the newly created vocabulary's endpoint is reachable.
+    // @todo figure out why we need to rebuild caches for it to be available.
+    drupal_flush_all_caches();
+    $response = $this->request('/jsonapi/taxonomy_term/im_a_vocab');
+    $this->assertEquals(200, $response->getStatusCode());
+
+    $description = 'How quickly deft jumping zebras vex.';
+    $data = [
+      'data' => [
+        'type' => 'taxonomy_term--im_a_vocab',
+        'id' => 'zebra_taxonomy_term',
+        'attributes' => [
+          'name' => 'zebra',
+          'uuid' => 'zebra_taxonomy_term',
+          'description' => [
+            'value' => $description,
+            'format' => 'rich_text',
+          ]
+        ],
+        'relationships' => [
+          'vid' => [
+            'data' => [
+              'type' => 'taxonomy_vocabulary--taxonomy_vocabulary',
+              'id' => 'taxonomy_test_vocabulary',
+            ]
+          ]
+        ]
+      ]
+    ];
+
+    // Create a taxonomy term (content entity).
+    $this->request('/jsonapi/taxonomy_term/im_a_vocab', 'post', $this->token, $data);
```

Comment:
Where is this method coming from? I don't see where we define it?

Reply:
`request` is defined in ApiTestBase.